lib/index.js

```diff
@@ -553,4 +554,93 @@ function equals(mat, operand) {
     }
 
     return op1.every((val, ind1) => val.every((ele, ind2) => Math.abs(ele - op2[ind1][ind2]) < 1e-10));
-}   
+}
+
+
+/**
+ * Calculates the adjunct matrix
+ *
+ * @param mat input matrix that initialized the function
+ * @returns adjunct matrix
+ */
+
+ function adjunct(mat) {
+    const [rows, cols] = size(mat)
+
+    const submtx = [] //variable to store the sub matrixes
+    const confactors = [] //variable to store the cofactors array
+
+    let index = () => { //function to return the indexs
+        let idx = []
+        for (let n = 0; n < rows * cols; n++) {
+            idx.push([])
+            for (let c = 0; c < cols; c++) {
+                idx[n].push(c)
+            }
+
+        }
+        return idx
+    }
+
+    let index_cols = () => {
+        const ic = index()
+        const [icRows, icCols] = size(ic)
+        let i = 0
+        while (i < icRows) {
+            for (let c = 0; c < icCols; c++) {
+                ic[i].splice(c, 1)
+                i++
+            }
+        }
+        return ic
+    }
+
+    let index_rows = () => {
+        const ir = index()
+        let n = 0
+        for (let l = 0; l < cols; l++) {
+            for (let c = 0; c < rows; c++) {
+                ir[n].splice(l, 1)
+                n++
+            }
+        }
+
+        return ir
+    }
+
+    for (let n = 0; n < rows * cols; n++) {
+        submtx.push([])
+        for (let l = 0; l < rows - 1; l++) {
+            submtx[n].push([])
+            for (let c = 0; c < cols - 1; c++) {
+                submtx[n][l].push(mtx
+                [index_rows()[n][l]]
+                [index_cols()[n][c]]
+                )
+            }
+        }
+    }
+
+    //get the cofactor of submtx
+    let n = 0
+    for (let l = 0; l < rows; l++) {
+        for (let c = 0; c < cols; c++) {
+            confactors.push(cofactor(l + 1, c + 1, submtx[n]))
+            n++
+        }
+    }
+
+    //transform the confactors array in a matrix
+    let mtx_confactors = []
+    for (let i = 0; i < rows; i++) {
+        mtx_confactors.push(confactors.slice(0, cols))
+        confactors.splice(0, cols)
+    }
+
+    return trans(mtx_confactors)
+}
+
+function cofactor(rows, cols, submtx) {
+    return (-1) ** (rows + cols) * determinant(submtx)
```

Comment:
FYI, the existing function uses row elimination to compute the determinant.

Reply:
Do I have to use ````matrix(submtx).det()```` instead of ````determinant(submtx)````?
I don't understand very well 😶

Reply:
Did you mean gaussian elimination?

Reply:
I think you can continue to use `determinant(submtx)` as it is an internal reference. All what I meant was in computing cofactors we are using a determinant function implemented already using row elimination. I think it is fine. Otherwise, you have compute it recursively which makes it more expensive.

Reply:
oh ok now I understand